### ISSUE TITLE
Hipchat connection times out after a while

### ISSUE THREAD

**Issue by beezly** [hipchat/hubot-hipchat#111]
I keep having a problem where the XMPP connection to hipchat gets timed out. 

The debug log produces the following output:

``````
[Sat Jun 15 2013 17:02:49 GMT+0100 (BST)] DEBUG  OUT >
[Sat Jun 15 2013 17:03:22 GMT+0100 (BST)] DEBUG  OUT >
[Sat Jun 15 2013 17:03:52 GMT+0100 (BST)] DEBUG  OUT >
[Sat Jun 15 2013 17:04:23 GMT+0100 (BST)] DEBUG  OUT >
[Sat Jun 15 2013 17:04:54 GMT+0100 (BST)] DEBUG  OUT >
[Sat Jun 15 2013 17:05:25 GMT+0100 (BST)] DEBUG  OUT >
[Sat Jun 15 2013 17:05:56 GMT+0100 (BST)] DEBUG  OUT >
[Sat Jun 15 2013 17:06:33 GMT+0100 (BST)] DEBUG  OUT >
[Sat Jun 15 2013 17:11:37 GMT+0100 (BST)] INFO Disconnected from hipchat.com
[Sat Jun 15 2013 17:11:37 GMT+0100 (BST)] ERROR 'service-unavailable', 'Timed out after 150s of inactivity', { name: 'stream:error',
  parent: null,
  attrs: { 'xmlns:stream': 'http://etherx.jabber.org/streams' },
  children:
   [ { name: 'service-unavailable',
       parent: [Circular],
       attrs: { xmlns: 'urn:ietf:params:xml:ns:xmpp-streams' },
       children: [] },
     { name: 'text',
       parent: [Circular],
       attrs: { xmlns: 'urn:ietf:params:xml:ns:xmpp-streams' },
       children: [ 'Timed out after 150s of inactivity' ] } ] }
[Sat Jun 15 2013 17:11:37 GMT+0100 (BST)] DEBUG   IN > <stream:error><service-unavailable xmlns='urn:ietf:params:xml:ns:xmpp-streams'/><text xmlns='urn:ietf:params:xml:ns:xmpp-streams'>Timed out after 150s of inactivity</text></stream:error></stream:stream>

/home/cs1ajb/cicsbot/node_modules/hubot-hipchat/node_modules/node-xmpp/lib/xmpp/connection.js:277
    this.socket.end();
                ^
TypeError: Cannot call method 'end' of undefined
  at Client.Connection.onEnd (/home/cs1ajb/cicsbot/node_modules/hubot-hipchat/node_modules/node-xmpp/lib/xmpp/connection.js:277:17)
  at CleartextStream.<anonymous> (/home/cs1ajb/cicsbot/node_modules/hubot-hipchat/node_modules/node-xmpp/lib/xmpp/connection.js:56:14)
  at CleartextStream.EventEmitter.emit (events.js:117:20)
  at _stream_readable.js:910:16
  at process._tickCallback (node.js:415:13)```
``````

**Comment by grandmasterspoc** [comment]
It's been 2 years since this was open.. updates please?

**Comment by rbergman** [comment]
A reconnection impl was merged with PR #174, and improved in some subsequent PRs.  node-xmpp was also recently updated.
